### PR TITLE
feat: show manifests in image list

### LINKED ISSUE #6529: Manifest feature: List manifests on image list screen
### Is your feature request related to a problem? Please describe

When you create a manifest on the CLI:

```
podman manifest create test123
```

It appears on `podman image ls`:

```
▶ podman image ls
REPOSITORY                                TAG                IMAGE ID      CREATED       SIZE
localhost/test123                         latest             8b9101797ef4  2 days ago    1.08 kB
localhost/foobar                          latest             409f1927cb03  3 days ago    112 B
quay.io/podai/test123                     latest             e721af43292a  3 days ago    1.64 GB
```

However, it does not appear within the Podman Desktop UI.

### Describe the solution you'd like

Match the podman CLI behaviour by showing the manifest on the image list. 

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### LINKED ISSUE #7255: Manifest basic "details" page
### Is your feature request related to a problem? Please describe

Similar to how we click on images, pods, containers. etc and we get the details page. The same should be done with manifest images.

### Describe the solution you'd like

Details on manifest images (inspect, history, etc.) when clicking on the image.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by cdrage**
feat: show manifests in image list

### What does this PR do?

* Shows manifest in the image list
* Does NOT show any actions / be able to click (yet)
* Allows you to click and view details.

### Screenshot / video of UI

https://github.com/containers/podman-desktop/assets/6422176/e930329d-2a84-4c2b-b719-373f12f0b22c

### What issues does this PR fix or reference?

Closes https://github.com/containers/podman-desktop/issues/6529
Closes https://github.com/containers/podman-desktop/issues/7255

### How to test this PR?

- [X] Tests are covering the bug fix or the new feature

1. Create a manifest through the GUI or `podman build --platform
   linux/arm64,linux/amd64 --manifest testmanifest123`
2. View the manifest in image list.

Signed-off-by: Charlie Drage <charlie@charliedrage.com>

### CODE REVIEW

**Comment by cdrage** [comment]
FYI it is intentional that there are no actions shown / unable to click for more details / no delete button. As those are in separate PR's in the epic: https://github.com/containers/podman-desktop/issues/6526 specifically issues: #7041 and #7181

File: packages/renderer/src/lib/image/image-utils.ts
Comment (cdrage) [inline packages/renderer/src/lib/image/image-utils.ts]: unsure which colour we should use to indicate manifest, so just chose bg-purple-600 for now @deboer-tim @benoitf

**Review by deboer-tim — CHANGES_REQUESTED** [review]
I think it's curious/annoying that manifests are images in the API, and makes it so that something like ImageColumnActions needs to be shared. :-/

I think this is a good place to go with the first PR, but:
- In the containers view we have "my-pod (pod)", but here you used a badge. We obviously didn't have badges when we did the containers view, but I think we need to decide with design the right way to identify 'different things in a table' is and make them consistent. For simplicity/consistency maybe I'd rather use '(manifest)' until we know.
- The icon should be different from a regular image.
- Manifests should have details pages (but can be a separate issue).
- We should allow them to expand/collapse with the images that are in them, but that is post #7209.

I don't want to hold this up too much, but I think we should at least have a direction on 1/2 and issue for 3 before we merge.

**Comment by cdrage** [comment]
> I think it's curious/annoying that manifests are images in the API, and makes it so that something like ImageColumnActions needs to be shared. :-/
> 
> I think this is a good place to go with the first PR, but:
> 
> * In the containers view we have "my-pod (pod)", but here you used a badge. We obviously didn't have badges when we did the containers view, but I think we need to decide with design the right way to identify 'different things in a table' is and make them consistent. For simplicity/consistency maybe I'd rather use '(manifest)' until we know.
> * The icon should be different from a regular image.
> * Manifests should have details pages (but can be a separate issue).
> * We should allow them to expand/collapse with the images that are in them, but that is post [Table component support for parent/children #7209](https://github.com/containers/podman-desktop/issues/7209).
> 
> I don't want to hold this up too much, but I think we should at least have a direction on 1/2 and issue for 3 before we merge.

I totally agree, let's talk about it in the UX call. There are a lot of other missing pieces that should be worked on before we merge the "showing in list" support. 

For this PR I will:

- [x] Add (manifest) similar to (my-pod) instead of a badge
- [ ] Different icon (will ask you or @ekidneyrh on advice what we should do)

I thought I had opened a "details" issue for manifest, but I had not. I've opened up issue https://github.com/containers/podman-desktop/issues/7255 and added it to the epic https://github.com/containers/podman-desktop/issues/6526

Maybe we should keep this as draft / WIP until we get the details page in too so we at least have the basic functionality of manifests.

**Comment by cdrage** [comment]
This PR is ready for review with the exception of the new icon being added, which can be in a follow up PR.

I've made the following changes:
- New folder / files, based on how we do ImageDetails.svelte
- Added a BASIC manifest summary page, a more advanced summary page based on using inspectManifest will be added via issue: https://github.com/containers/podman-desktop/issues/7270
- Added tests

File: packages/renderer/src/lib/image/ImageDetailsHistory.svelte
Comment (lstocchi) [inline packages/renderer/src/lib/image/ImageDetailsHistory.svelte]: to be removed

**Review by lstocchi — APPROVED** [review]
Tested and works as described. LGTM

**Comment by cdrage** [comment]
Manifest icon has now been added and this PR is ready for another review @lstocchi @deboer-tim

**Review by lstocchi — APPROVED** [review]

**Review by deboer-tim — APPROVED** [review]
LGTM, worked well in testing.

**Pull request merged**
